### PR TITLE
refactor: Streamline server config generation

### PR DESCRIPTION
**Pull request by atomrc**
This will:
- strongly type frontend config
- make sure process.env contains at least some mandatory fields
- stop writing commit and version files to the filesystem
- enforce strong types for the general app config

### CODE REVIEW

File: src/script/Config.ts [hunk 1]
```diff
@@ -31,110 +31,65 @@ export const ACCENT_ID = {
...
-  readonly APP_NAME = env.APP_NAME || 'Webapp';
-  readonly APP_INSTANCE_ID = createRandomUuid();
-  readonly BACKEND_NAME = env.BACKEND_NAME || 'Wire';
-  readonly BACKEND_REST = env.BACKEND_REST || 'https://prod-nginz-https.wire.com';
-  readonly BACKEND_WS = env.BACKEND_WS || 'wss://prod-nginz-ssl.wire.com';
-  readonly BRAND_NAME = env.BRAND_NAME || 'Wire';
-  readonly COUNTLY_API_KEY = env.COUNTLY_API_KEY;
-  readonly ENABLE_DEV_BACKEND_API = env.ENABLE_DEV_BACKEND_API;
-  readonly ENVIRONMENT = env.ENVIRONMENT || 'production';
-  readonly FEATURE = {
```
Comment (atomrc) [inline src/script/Config.ts:45]: Note that now all the types and default values are generated server side. No need for this huge config file anymore

File: src/script/components/Conversation/Conversation.tsx [hunk 2]
```diff
@@ -187,7 +187,7 @@ export const Conversation: FC<ConversationProps> = ({
 
       if (!isHittingUploadLimit(droppedFiles, repositories.asset)) {
         Array.from(droppedFiles).forEach(file => {
-          const isSupportedImage = CONFIG.ALLOWED_IMAGE_TYPES.includes(file.type);
+          const isSupportedImage = (CONFIG.ALLOWED_IMAGE_TYPES as ReadonlyArray<string>).includes(file.type);
```
Comment (atomrc) [inline src/script/components/Conversation/Conversation.tsx:190]: This is needed now that the config is a `as const` (see https://stackoverflow.com/a/56745484)

File: src/types/Wire.types.ts [hunk 3]
```diff
@@ -17,86 +17,10 @@
  *
  */
 
+import {ClientConfig} from '../../server/config';
 import {App} from '../script/main/app';
 
-type FeatureVariables = {
```
Comment (atomrc) [inline src/types/Wire.types.ts:22]: No more manual typing of this 😎  It's automatically inferred from the client config on the server side

File: server/package.json [hunk 4]
```diff
@@ -35,10 +35,8 @@
     "typescript": "5.0.4"
   },
   "scripts": {
-    "build": "yarn clean && tsc && yarn copy-assets && yarn create-commit-sha && yarn create-version-file",
+    "build": "yarn clean && tsc && yarn copy-assets",
     "copy-assets": "node ./bin/copy_server_assets.js",
-    "create-commit-sha": "node ./bin/create_commit_sha_file.js",
-    "create-version-file": "node ./bin/create_version_file.js",
```
Comment (atomrc) [inline server/package.json:41]: No more generating static files for this, this is computed in the config and served from memory when the endpoint `/commit` or `/version` are hit

File: server/Server.ts [hunk 5]
```diff
@@ -205,16 +202,14 @@ class Server {
     this.app.set('views', [path.resolve(__dirname, 'static'), path.resolve(__dirname, 'templates')]);
     hbs.localsAsTemplateData(this.app);
     this.app.locals.config = {
-      APP_BASE: this.config.SERVER.APP_BASE,
-      BRAND_NAME: this.config.CLIENT.BRAND_NAME,
-      CHROME_ORIGIN_TRIAL_TOKEN: this.config.CLIENT.CHROME_ORIGIN_TRIAL_TOKEN,
```
Comment (atomrc) [inline server/Server.ts:210]: This was very old way to enable some custom features on Chrome :) 
Not used anymore :)

File: src/script/Config.ts [hunk 6]
```diff
@@ -17,9 +17,9 @@
  *
  */
 
-import {ValidationUtil, Runtime} from '@wireapp/commons';
+import {Runtime} from '@wireapp/commons';
 
-import {createRandomUuid} from 'Util/util';
+import {createUuid} from 'Util/uuid';
```
Comment (atomrc) [inline src/script/Config.ts:22]: This was a cyclic dependency (`Utils` imports `Config` and vice-versa) 🙃

**Review by PatrykBuniX — APPROVED** [review]

**Pull request merged**